app/controllers/RemoveNiphlController.scala

```diff
@@ -63,11 +65,25 @@ class RemoveNiphlController @Inject() (
         .fold(
           formWithErrors => Future.successful(BadRequest(view(formWithErrors))),
           value =>
-            request.userAnswers.set(RemoveNiphlPage, value) match {
-              case Success(answers) =>
-                sessionRepository.set(answers).map { _ =>
-                  Redirect(navigator.nextPage(RemoveNiphlPage, NormalMode, answers))
-                }
+            if (value) {
```

Comment:
this was a bug that i found whilst testing different scenarios 

basically the bug was only present on niphl, not nirms so to fix i duplicated what was in the nirms journey so now the code is the same for both